repo: add documentation (#12)
NordicSemiconductor/npmx-zephyr: doc/index.rst

.. _index:

> Welcome to npmx zephyr PMIC support repository!

Reviewer: ```suggestion
Welcome to `npmx-zephyr` repository!
```

###############################################

> :ref:`npmx-zephyr <index>` is a repository with zephyr-like samples for Nordic Semiconductor `npmx library`_ and driver integration with `zephyr`_ environment.

Reviewer: ```suggestion
:ref:`npmx-zephyr <index>` is a repository showcasing the integration of `npmx library`_ into an RTOS environment, using `Zephyr`_ RTOS as an example.
```
Is it ok to use `Zephyr`? Will the link still work?

Author: `Zephyr` works good

> If you want to see how samples work in zephyr, see `Samples and Demos`_ section.

Reviewer: ```suggestion
If you want to see how samples work in Zephyr, see `Samples and Demos`_ section.
```

.. _index:

Welcome to npmx zephyr PMIC support repository
##############################################

Nordic Semiconductor announces multi-function `Power Management`_ IC (PMIC) with unique system management features.

> :ref:`npmx-zephyr <index>` is a repository with zephyr-like samples for Nordic Semiconductor `npmx library`_.

Reviewer: Also, driver integration.

Zephyr also provides a variety of application samples and demos.
Documentation for those is available in Zephyr's `Samples and Demos`_ section.
For very simple samples, see the `Basic Samples`_.

> Those samples are a good starting point for understanding how to put together your own application.

Reviewer: No need to explain what other zephyr examples. Just linking to Zephyr's getting started would be ok.
